File: BookFiles/Strategy/Code/Strategy_Main.java
public class Strategy_Main {

  public static void main(String[] args) {
    
    System.out.println("A tiny frigate wants some trouble");
    BattleShip battleShip = new BattleShip(new BruteForceStrategy());
    battleShip.engage();
    System.out.println("Four tiny frigates want some trouble");
    battleShip.alterManeuvers(new DivideConquerStrategy());
    battleShip.engage();
    System.out.println("An aircraft carrier group wants some trouble");
    battleShip.alterManeuvers(new RetreatStrategy());
    battleShip.engage();
  }
}

File: BookFiles/Strategy/C#_Code/Strategy_Main.cs
using System;

namespace Strategy
{
    class Strategy_Main
    {
        static void Main(string[] args)
        {
            Console.WriteLine("A tiny frigate wants some trouble");
            BattleShip battleShip = new BattleShip(new BruteForceStrategy());
            battleShip.engage();
            Console.WriteLine("Four tiny frigates want some trouble");
            battleShip.alterManeuvers(new DivideConquerStrategy());
            battleShip.engage();
            Console.WriteLine("An aircraft carrier group wants some trouble");
            battleShip.alterManeuvers(new RetreatStrategy());
            battleShip.engage();
        }
    }
}
